scripts/components/dependencies_validator.test.ts

```diff
@@ -142,6 +144,33 @@ void describe('Dependency validator', () => {
     );
   });
 
+  void it('can detect inconsistent major versions of repo packages', async () => {
```

Comment:
Should we also add a test that inconsistent non-major versions of repo packages are ok?